go-controller/pkg/node/port_claim.go

```diff
@@ -70,41 +87,65 @@ func initPortClaimWatcher(recorder record.EventRecorder, wf factory.NodeWatchFac
 			}
 		},
 	}, nil)
+	return nil
 }
 
 func addServicePortClaim(svc *kapi.Service) []error {
-	return handleService(svc, port.open)
+	return handleService(svc, portHandler.open)
 }
 
 func deleteServicePortClaim(svc *kapi.Service) []error {
-	return handleService(svc, port.close)
+	return handleService(svc, portHandler.close)
 }
 
 func handleService(svc *kapi.Service, handler handler) []error {
 	errors := []error{}
 	if !util.ServiceTypeHasNodePort(svc) && len(svc.Spec.ExternalIPs) == 0 {
 		return errors
 	}
+
 	for _, svcPort := range svc.Spec.Ports {
 		if util.ServiceTypeHasNodePort(svc) {
-			if err := handlePort(svc, svcPort.NodePort, svcPort.Protocol, handler); err != nil {
+			klog.V(5).Infof("Handle NodePort service %s port %d", svc.Name, svcPort.NodePort)
+			if err := handlePort(getDescription(svcPort.Name, svc, true), svc, "", svcPort.NodePort, svcPort.Protocol, handler); err != nil {
 				errors = append(errors, err)
 			}
 		}
-		if len(svc.Spec.ExternalIPs) > 0 {
-			if err := handlePort(svc, svcPort.Port, svcPort.Protocol, handler); err != nil {
+		for _, externalIP := range svc.Spec.ExternalIPs {
+			klog.V(5).Infof("Handle ExternalIPs service %s external IP %s port %d", svc.Name, externalIP, svcPort.Port)
+			if err := handlePort(getDescription(svcPort.Name, svc, false), svc, externalIP, svcPort.Port, svcPort.Protocol, handler); err != nil {
 				errors = append(errors, err)
 			}
 		}
 	}
 	return errors
 }
 
-func handlePort(svc *kapi.Service, port int32, protocol kapi.Protocol, handler handler) error {
+// LocalPorts allows to add an arbitrary description, which can be used to distinguish LocalPorts instances having the
+// same networking parameters by created for different services.
+// kube-proxy and this implementation use the following format of the description: "
+//        for NodePort services            - "nodePort for namespace/name[:portName]
+//        for services with External IPs   - "externalIP for namespace/name[:portName]
+func getDescription(portName string, svc *kapi.Service, nodePort bool) string {
+	svcName := types.NamespacedName{Namespace: svc.Namespace, Name: svc.Name}
+	var prefix string
```

Comment:
nit: there is no need to branch; you can start off by assigning prefix to one of the descriptor prefixes like so:

```
prefix := externalPortDescr
if nodePort {
      prefix = nodePortDescr
}
```

Reply:
Done